scripts/release/verify-artifacts.py

```
@@ -0,0 +1,519 @@
+        raise VerificationError("artifact manifest has an invalid structure")
+    if manifest["schema_version"] != 1 or not isinstance(manifest["artifacts"], list):
+        raise VerificationError("artifact manifest has an unsupported schema")
+
+    entries: dict[str, str] = {}
+    for entry in manifest["artifacts"]:
+        if not isinstance(entry, dict) or set(entry) != {"path", "sha256"}:
+            raise VerificationError("artifact manifest entry has an invalid structure")
+        path = entry["path"]
+        digest = entry["sha256"]
+        if not isinstance(path, str) or path in entries:
+            raise VerificationError(f"duplicate artifact manifest path: {path}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)